code/datums/components/z_parallax.dm

```diff
@@ -0,0 +1,60 @@
+/**
+ * Component that hooks into the client, listens for COMSIG_MOVABLE_Z_CHANGED, and depending on whether or not the
+ * Z-level has ZTRAIT_NOPARALLAX enabled, disable or reenable parallax.
+ */
+
+/datum/component/zparallax
```

Comment:
for the record, it might in theory be better to have this component parent off client, since you never actually use the parent (IE: the hud datum) in this code.
It's not a huge deal, and that inconsistency could be resolved with a change on line 58 (and a typecheck in initialize), which ya don't really need to do, but I figured I'd mention it

Reply:
client changes based on if you reconnect, but hud is the exact same. id reckon it would probably be best to put this on the client to avoid remaking this component everytime a player reconnects